Microsoft.Toolkit.Services/Services/MicrosoftGraph/MicrosoftGraphService.cs

```diff
@@ -193,15 +193,20 @@ public virtual Task<bool> Logout()
                 throw new InvalidOperationException("Microsoft Graph not initialized.");
             }
 
-            IsAuthenticated = false;
             User = null;
 
+            Task<bool> result;
+
 #if WINRT
             var authenticationModel = AuthenticationModel.ToString();
-            return Authentication.LogoutAsync(authenticationModel);
+            result = Authentication.LogoutAsync(authenticationModel);
 #else
-            return Task.Run(() => { return Authentication.Logout(); });
+            result = Task.Run(() => { return Authentication.Logout(); });
 #endif
+
+            IsAuthenticated = false;
```

Comment:
Shouldn't we wait for the result task to complete before seting the flag to false?

Reply:
@nmetulev Fixed, I meant to use the pattern `result.ContinueWith((o) => IsAuthenticated = false);`, however, that might break some existing integration which update the UI elements in the event `IsAuthenticatedChanged` like the [AadAuthControl](https://github.com/Microsoft/WindowsCommunityToolkit/blob/d6493c7176a065a405dfce8915a29031f8564619/Microsoft.Toolkit.Uwp.SampleApp/Controls/AadAuthControl.xaml.cs#L68), let me know if you have a different opinion.

```
The application called an interface that was marshalled for a different thread. (Exception from HRESULT: 0x8001010E (RPC_E_WRONG_THREAD))
```

Reply:
You might be able to specify that you want ContinueWith to capture the current thread context and continue on the same thread: https://stackoverflow.com/questions/18312695/does-task-continuewith-capture-the-calling-thread-context-for-continuation

Reply:
@nmetulev Thanks for the advice, I should google first. The change is already done and validated.